pkg/provider/data_source_pgd.go

```diff
@@ -0,0 +1,314 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/api"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/models/pgd"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/utils"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ datasource.DataSource              = &pgdDataSource{}
+	_ datasource.DataSourceWithConfigure = &pgdDataSource{}
+)
+
+type pgdDataSource struct {
+	client *api.API
+}
+
+func (p pgdDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pgd"
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *pgdDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	d.client = req.ProviderData.(*api.API)
```

Comment:
After creating a specific client (see my comment in the read method of this data source), can you set that client here? Something like:
```
d.client = req.ProviderData.(*api.API).PgdClient()
```

so that we can call the client functions of the PGDClient within the data source methods.